exporter/datadogexporter/internal/clientutil/api.go

```diff
@@ -15,13 +15,70 @@
 package clientutil // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/clientutil"
 
 import (
+	"context"
 	"errors"
+	"os"
 
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV1"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
 	"go.uber.org/zap"
 	zorkian "gopkg.in/zorkian/go-datadog-api.v2"
 )
 
+// CreateAPIClient creates a new Datadog API client
+func CreateAPIClient(buildInfo component.BuildInfo, endpoint string, settings exporterhelper.TimeoutSettings, insecureSkipVerify bool) *datadog.APIClient {
+	configuration := datadog.NewConfiguration()
+	configuration.UserAgent = UserAgent(buildInfo)
+	configuration.HTTPClient = NewHTTPClient(settings, insecureSkipVerify)
+	configuration.Compress = true
+	if endpoint != "" {
```

Comment:
When would we get `endpoint == ""`? What should happen in that situation?

Reply:
If left unspecified, the endpoint will be set based on the site https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e46477ef0700b313e7404d42b51f5bcab10273d2/exporter/datadogexporter/config.go#L491-L500, so I don't think we would get this in practice.

I have a slight preference to make the whole thing fail if the endpoint is empty, but since I don't feel strongly I will approve this and merge after you see this.

Reply:
Sounds good, I'd prefer to do the validation in a follow-up PR since that's a fairly independent change.